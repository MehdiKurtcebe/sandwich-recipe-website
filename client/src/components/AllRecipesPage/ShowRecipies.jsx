import React from 'react';
import FavoriteIcon from '@material-ui/icons/Favorite';
import IconButton from '@material-ui/core/IconButton';
import { useStore } from '../../contextAPI/store/Provider';
import {
    notifyWarnRight,
    notifySuccessRight,
    notifyInfoRight
} from '../Error_Page/Messages';
import UserServices from '../../services/UserServices';
import { setUSer } from '../../contextAPI/actions/LoginAction';
import {
    Button,
    Card,
    CardImg,
    CardText,
    CardBody,
    Col,
    Container,
    Row
} from 'reactstrap';
const Recipies = ({ recipies }) => {
    let len = 4;
    if (
        recipies &&
        recipies.length === 1 &&
        recipies[0]._id === '5fb967aed21567abd722a076'
    )
        len = 14;
    const [{ isLogged, user }, dispatch] = useStore('');
    const addToFav = id => {
        if(isLogged){
            if(!checkContain(id)){
                user.favoriteRecipes.push(id);
                UserServices.update(user._id,user)
                .then(response => {
                    dispatch(setUSer(user));
                })
                .catch(e => {
                    console.log(e);
                });
                notifySuccessRight("Recipe added to your favorite recipes!");
            }
            else{
                notifyWarnRight("This recipe already added to favorites!");
            }
        }
        else{
            notifyInfoRight("To add favorites Sign up or Login!");
        }
    };
    const checkContain = id => {
        return user.favoriteRecipes.includes(id);
    };
    return (
        <div className="container">
            <Container>
                <Row className="row">
                    {recipies.map((recipie, index) => {
                        return (
                            <Col md={len} key={index}>
                                <Card className="mb-4 box-shadow">
                                    <CardImg
                                        top
                                        width="180"
                                        height="250"
                                        src={recipie.imgURL}
                                    />
                                    <CardBody className="text-center">
                                        {/*THERE WILL AN item OBJECT IN DATABASE AND IT WILL BE PRINTED IN HERE*/}
                                        <CardText className="cardText">
                                            <strong className="strong">
                                                {recipie.name}
                                            </strong>
                                        </CardText>

                                        <div className="d-flex justify-content-between align-items-center">
                                            <Button
                                                className="showButton"
                                                href={`/recipe/${recipie._id}`}
                                                variant="outline-dark"
                                                color="secondary"
                                                size="lg"
                                                block
                                            >
                                                <strong>Show</strong>
                                            </Button>
                                            <IconButton
                                                color="secondary"
                                                aria-label="add to favorites"
                                                onClick={() => {
                                                    addToFav(recipie._id);
                                                }}
                                            >
                                                <FavoriteIcon />
                                            </IconButton>
                                        </div>
                                    </CardBody>
                                </Card>
                            </Col>
                        );
                    })}
                </Row>
            </Container>
        </div>
    );
};

export default Recipies;
