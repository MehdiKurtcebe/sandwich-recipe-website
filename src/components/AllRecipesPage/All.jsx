import RecipieDataService from '../../services/RecipieService';
import RecipeList from './RecipeList';
import AlRecipies from './AllRecipies';
import React, { Component } from 'react';

class All extends Component {
    state = {
        brands: [],
        recipies:[],
        selectedCheckboxes: new Set(),
        current_recipies: []
    };
    retrieveBrands = () => {
        RecipieDataService.getCategories()
            .then(response => {
                this.setState({ brands: response.data });
                console.log(response.data);
            })
            .catch(e => {
                console.log(e);
            });
    };
    retrieveRecipies = () => {
        RecipieDataService.getAll()
            .then(response => {
                this.setState({ recipies: response.data });
                this.setState({ current_recipies: response.data });
                console.log(response.data);
            })
            .catch(e => {
                console.log(e);
            });
    };
    componentDidMount() {
        this.retrieveBrands();
        this.retrieveRecipies();
        this.setState({ selectedCheckboxes: new Set() });;
    }
    handleSelectBox = e => {
        const name = e.target.name;
        const value = e.target.checked;
        if (this.state.selectedCheckboxes.has(name)) {
            this.state.selectedCheckboxes.delete(name);
        } else {
            this.state.selectedCheckboxes.add(name);
        }
        this.setState({ selectedBoxes: this.state.selectedCheckboxes });
        if (this.state.selectedCheckboxes.size != 0) {
            this.setRecipies();
        } else {
            this.setState({ current_recipies: this.state.recipies });
        }
    };
    setRecipies = () => {
       let temp=[]
       for (let item of this.state.selectedCheckboxes){
            for (let rec of this.state.recipies){
                if(rec.category.includes(item) && !temp.includes(rec)){
                    temp.push(rec)
                }
            }
       }
        this.setState({ current_recipies: temp });
    };
    render() {
        return (
            <div
                className="container"
                style={{ paddingTop: '2rem', paddingLeft: '0rem' }}
            >
                <div className="row" style={{ marginLeft: '-5rem' }}>
                    <AlRecipies
                        brands={this.state.brands}
                        handleSelectBox={this.handleSelectBox}
                    />
                    <RecipeList
                        recipies={this.state.current_recipies}
                    />
                </div>
            </div>
        );
    }
}

export default All;
