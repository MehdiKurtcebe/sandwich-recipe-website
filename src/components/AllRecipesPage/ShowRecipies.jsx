import React from 'react';
import {
    Button,
    Card,
    CardImg,
    CardText,
    CardBody,
    Col,
    Container,
    Row
} from 'reactstrap';
const Album = ({ recipies }) => {
    return (
        <div className="recipies py-5 bg-light">
            <Container>
                <Row>
                    {recipies.map((recipie, index) => {
                        return (
                            <Col md="4" key={index}>
                                <Card className="mb-4 box-shadow">
                                    <CardImg
                                        top
                                        width="180" 
                                        height="250"
                                        src={recipie.imgURL}
                                    />
                                    <CardBody className="text-center">
                                        {/*THERE WILL AN item OBJECT IN DATABASE AND IT WILL BE PRINTED IN HERE*/}
                                        <CardText><strong >{recipie.name}</strong></CardText>

                                        <div className="d-flex justify-content-between align-items-center">
                                            <Button
                                                href={`/recipe/${recipie._id}`}
                                                variant = "outline-dark"
                                                color="secondary"
                                                size="lg"
                                                block
                                            >
                                            <strong>Show</strong>
                                            </Button>
                                        </div>
                                    </CardBody>
                                </Card>
                            </Col>
                        );
                    })}
                </Row>
            </Container>
        </div>
    );
};

export default Album;
