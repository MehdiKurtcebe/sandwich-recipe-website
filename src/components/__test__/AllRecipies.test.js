import React from 'react';
import ReactDOM from 'react-dom';
import All from '../AllRecipesPage/AllRecipies';
import Filter from '../AllRecipesPage/FilterBox';
import RecipieList from '../AllRecipesPage/RecipeList';
import ShowRecipies from '../AllRecipesPage/ShowRecipies';
import Enzyme from 'enzyme';
//import chai, {expect} from 'chai';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount,shallow } from 'enzyme';
import RecipieDataService from '../../services/RecipieService';
import { Button } from 'reactstrap';

Enzyme.configure({ adapter: new Adapter() });
describe('All Recipies Page Component Tests', () => {
    var brands = [];
    var recipes = [];
    let wrapper;
    const retrieveBrands = () => {
        RecipieDataService.getCategories()
        .then(response => {
            brands.join(response.data);
            console.log(response.data);
        })
        .catch(e => {
            console.log(e);
        });
    };
    const retrieveRecipies = () => {
        RecipieDataService.getAll()
            .then(response => {
                recipes.join(response.data);
                console.log(response.data);
            })
            .catch(e => {
                console.log(e);
            });
    };
    beforeEach(() => {
        retrieveBrands();
        retrieveRecipies();
    });
    it("renders without crashing", () => {
        const component = mount(<Filter brands={brands}/>);
        expect(component).toMatchSnapshot();
      });
    it('renders checkbox name', () => {
        wrapper = shallow(<Filter brands={brands}/>)
        expect(wrapper.find('input.custom-checkbox__input')).exist;
    });
    it('renders filter title', () => {
        wrapper = shallow(<Filter brands={brands}/>)
        expect(wrapper.find('h3.filter_title').text()).toEqual("Categories");
    });
    it('renders row', () => {
        wrapper = shallow(<ShowRecipies recipies={recipes}/>);
        expect(wrapper.find('Row.row')).exist;
    });
    it('renders card', () => {
        wrapper = shallow(<ShowRecipies recipies={recipes}/>);
        expect(wrapper.find('Card.mb-4 box-shadow')).exist;
    });
    it('renders recipe name', () => {
        wrapper = shallow(<ShowRecipies recipies={recipes}/>);
        expect(wrapper.find('CardText.cardText')).exist;
    });
    it('goes to /recipe/${recipie._id} when clicked', () => {
        wrapper = shallow(<ShowRecipies recipies={recipes}/>);
        expect(wrapper.find("Button.showButton")).exist;
    })

});
