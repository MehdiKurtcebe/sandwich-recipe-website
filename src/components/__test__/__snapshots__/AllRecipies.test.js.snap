// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Recipies Page Component Tests renders without crashing 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
    "wrapInvoke": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "brands": Array [],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="col-lg-3"
        style="padding-right: 3rem;"
      >
        <div
          class="row"
        >
          <div
            class="col-15"
          >
            <div
              class="card mb-3"
            >
              <div
                class="card-header"
              >
                <h3
                  class="filter_title"
                >
                  Categories
                </h3>
              </div>
              <ul
                class="list-group flex-row flex-wrap"
                id="ul1"
              />
            </div>
          </div>
        </div>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-15"
          >
            <div
              className="card mb-3"
            >
              <div
                className="card-header"
              >
                <h3
                  className="filter_title"
                >
                  Categories
                </h3>
              </div>
              <ul
                className="list-group flex-row flex-wrap"
                id="ul1"
              />
            </div>
          </div>
        </div>,
        "className": "col-lg-3",
        "style": Object {
          "paddingRight": "3rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="row"
          >
            <div
              class="col-15"
            >
              <div
                class="card mb-3"
              >
                <div
                  class="card-header"
                >
                  <h3
                    class="filter_title"
                  >
                    Categories
                  </h3>
                </div>
                <ul
                  class="list-group flex-row flex-wrap"
                  id="ul1"
                />
              </div>
            </div>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-15"
            >
              <div
                className="card mb-3"
              >
                <div
                  className="card-header"
                >
                  <h3
                    className="filter_title"
                  >
                    Categories
                  </h3>
                </div>
                <ul
                  className="list-group flex-row flex-wrap"
                  id="ul1"
                />
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="col-15"
              >
                <div
                  class="card mb-3"
                >
                  <div
                    class="card-header"
                  >
                    <h3
                      class="filter_title"
                    >
                      Categories
                    </h3>
                  </div>
                  <ul
                    class="list-group flex-row flex-wrap"
                    id="ul1"
                  />
                </div>
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card mb-3"
                >
                  <div
                    className="card-header"
                  >
                    <h3
                      className="filter_title"
                    >
                      Categories
                    </h3>
                  </div>
                  <ul
                    className="list-group flex-row flex-wrap"
                    id="ul1"
                  />
                </div>,
                "className": "col-15",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <div
                    class="card mb-3"
                  >
                    <div
                      class="card-header"
                    >
                      <h3
                        class="filter_title"
                      >
                        Categories
                      </h3>
                    </div>
                    <ul
                      class="list-group flex-row flex-wrap"
                      id="ul1"
                    />
                  </div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="card-header"
                      >
                        <h3
                          className="filter_title"
                        >
                          Categories
                        </h3>
                      </div>,
                      <ul
                        className="list-group flex-row flex-wrap"
                        id="ul1"
                      />,
                    ],
                    "className": "card mb-3",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <div
                        class="card-header"
                      >
                        <h3
                          class="filter_title"
                        >
                          Categories
                        </h3>
                      </div>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h3
                          className="filter_title"
                        >
                          Categories
                        </h3>,
                        "className": "card-header",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": <h3
                            class="filter_title"
                          >
                            Categories
                          </h3>,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Categories",
                            "className": "filter_title",
                          },
                          "ref": null,
                          "rendered": Array [
                            "Categories",
                          ],
                          "type": "h3",
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": <ul
                        class="list-group flex-row flex-wrap"
                        id="ul1"
                      />,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [],
                        "className": "list-group flex-row flex-wrap",
                        "id": "ul1",
                      },
                      "ref": null,
                      "rendered": Array [
                        Array [],
                      ],
                      "type": "ul",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "brands": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="col-lg-3"
          style="padding-right: 3rem;"
        >
          <div
            class="row"
          >
            <div
              class="col-15"
            >
              <div
                class="card mb-3"
              >
                <div
                  class="card-header"
                >
                  <h3
                    class="filter_title"
                  >
                    Categories
                  </h3>
                </div>
                <ul
                  class="list-group flex-row flex-wrap"
                  id="ul1"
                />
              </div>
            </div>
          </div>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-15"
            >
              <div
                className="card mb-3"
              >
                <div
                  className="card-header"
                >
                  <h3
                    className="filter_title"
                  >
                    Categories
                  </h3>
                </div>
                <ul
                  className="list-group flex-row flex-wrap"
                  id="ul1"
                />
              </div>
            </div>
          </div>,
          "className": "col-lg-3",
          "style": Object {
            "paddingRight": "3rem",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="row"
            >
              <div
                class="col-15"
              >
                <div
                  class="card mb-3"
                >
                  <div
                    class="card-header"
                  >
                    <h3
                      class="filter_title"
                    >
                      Categories
                    </h3>
                  </div>
                  <ul
                    class="list-group flex-row flex-wrap"
                    id="ul1"
                  />
                </div>
              </div>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-15"
              >
                <div
                  className="card mb-3"
                >
                  <div
                    className="card-header"
                  >
                    <h3
                      className="filter_title"
                    >
                      Categories
                    </h3>
                  </div>
                  <ul
                    className="list-group flex-row flex-wrap"
                    id="ul1"
                  />
                </div>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <div
                  class="col-15"
                >
                  <div
                    class="card mb-3"
                  >
                    <div
                      class="card-header"
                    >
                      <h3
                        class="filter_title"
                      >
                        Categories
                      </h3>
                    </div>
                    <ul
                      class="list-group flex-row flex-wrap"
                      id="ul1"
                    />
                  </div>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card mb-3"
                  >
                    <div
                      className="card-header"
                    >
                      <h3
                        className="filter_title"
                      >
                        Categories
                      </h3>
                    </div>
                    <ul
                      className="list-group flex-row flex-wrap"
                      id="ul1"
                    />
                  </div>,
                  "className": "col-15",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <div
                      class="card mb-3"
                    >
                      <div
                        class="card-header"
                      >
                        <h3
                          class="filter_title"
                        >
                          Categories
                        </h3>
                      </div>
                      <ul
                        class="list-group flex-row flex-wrap"
                        id="ul1"
                      />
                    </div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          className="card-header"
                        >
                          <h3
                            className="filter_title"
                          >
                            Categories
                          </h3>
                        </div>,
                        <ul
                          className="list-group flex-row flex-wrap"
                          id="ul1"
                        />,
                      ],
                      "className": "card mb-3",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <div
                          class="card-header"
                        >
                          <h3
                            class="filter_title"
                          >
                            Categories
                          </h3>
                        </div>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <h3
                            className="filter_title"
                          >
                            Categories
                          </h3>,
                          "className": "card-header",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": <h3
                              class="filter_title"
                            >
                              Categories
                            </h3>,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Categories",
                              "className": "filter_title",
                            },
                            "ref": null,
                            "rendered": Array [
                              "Categories",
                            ],
                            "type": "h3",
                          },
                        ],
                        "type": "div",
                      },
                      Object {
                        "instance": <ul
                          class="list-group flex-row flex-wrap"
                          id="ul1"
                        />,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [],
                          "className": "list-group flex-row flex-wrap",
                          "id": "ul1",
                        },
                        "ref": null,
                        "rendered": Array [
                          Array [],
                        ],
                        "type": "ul",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <Recipies
    brands={Array []}
  />,
  Symbol(enzyme.__updatedBy__): null,
}
`;
